File: APCS/2406/p2.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(){
    int h, w, n;
    cin >> h >> w >> n;
    vector<vector<int>> arr(h, vector<int>(w, 0));

    for(int i = 0; i < n; i++){
        int r, c , t, num;
        cin >> r>> c >> t >> num;
        for(int j = 0; j < h; j++){
            for(int k = 0; k < w; k++){
                if(abs(j-r) + abs(k-c) < t+1) arr[j][k] += num;
            }
        }
    }
    
    for(auto row : arr){
        for(auto values : row){
            cout << values << " ";
        }
        cout << endl;
    }

    
}

File: APCS/2406/p2.py
h, w, n = map(int, input().split())
arr = [[0 for _ in range(w)] for _ in range(h)]
for i in range(n): 
    r,c , t, num = map(int, input().split())
    for j in range(h): 
        for k in range(w): 
            arr[j][k] += num if abs(j-r) + abs(k-c) <= t else 0

for row in arr: 
    for i, v in enumerate(row): print(f"{v}{ '\n' if i == w-1 else ''}", end='')
